allennlp/modules/attention/attention.py

```diff
@@ -32,38 +32,31 @@ class Attention(torch.nn.Module):
 
     Parameters
     ----------
-    similarity_function : ``SimilarityFunction``, optional (default=``DotProductSimilarity``)
-        The similarity function to use when computing the attention.
     normalize : ``bool``, optional (default: ``True``)
         If true, we normalize the computed similarities with a softmax, to return a probability
         distribution for your attention.  If false, this is just computing a similarity score.
     """
     def __init__(self,
-                 similarity_function: SimilarityFunction = None,
                  normalize: bool = True) -> None:
         super(Attention, self).__init__()
-
-        self._similarity_function = similarity_function or DotProductSimilarity()
         self._normalize = normalize
 
     @overrides
     def forward(self,  # pylint: disable=arguments-differ
```

Comment:
This is the base class, where we're overriding `torch.nn.Module.forward` directly, so we need to disable this pylint check here.  No subclasses of this class should disable this check.